tests/wpt/web-platform-tests/html/semantics/embedded-content/media-elements/event_canplay.html

```diff
@@ -14,25 +14,27 @@
   </video>
   <div id="log"></div>
   <script>
-test(function() {
+(function() {
   var t = async_test("setting src attribute on autoplay audio should trigger canplay event", {timeout:5000});
   var a = document.getElementById("a");
-  a.addEventListener("canplay", function() {
+  a.addEventListener("error", t.unreached_func());
+  a.addEventListener("canplay", t.step_func(function() {
```

Comment:
This could probably be `t.step_func_done(function() { a.pause() })` and similar below, but either way.  In any case, it's a bit weird that we pause _after_ claiming we're done.

I guess if you want to preserve that ordering, you really do need step_func here.